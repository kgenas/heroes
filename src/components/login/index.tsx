import React from "react";
import { useNavigate } from "react-router-dom"
import { useDispatch } from 'react-redux';
import { AppDispatch } from '../../app/store';
import { login } from "../../features/counter/loginHero";

export const Login = () =>{
    const navigate = useNavigate();
    const dispatch:AppDispatch = useDispatch();

    const handleLogion = () =>{

        const data = {
            name: 'pedro',                
        };
        dispatch(login(data));
        const lastPath = localStorage.getItem('lasPath') || '/marvel';
        navigate( lastPath, { replace: true });
    };
    return(
        <div className="container mt-5">
            <h1>Login</h1>
            <hr/>
            <button
                className="btn btn-primary"
                onClick={ handleLogion }
            >
                Login
            </button>
        </div>
    )    
}