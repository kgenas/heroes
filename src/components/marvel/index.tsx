import { HeroList ,HeroAll } from "../hero/HeroList";

export const MarvelScreen = () => {

    return (
        <div>
            <h1>MarvelScreen</h1>
            <hr/>
            <HeroList publisher ='Marvel Comics'/>
        </div>
    );
};

export const DCMarvelScreen = () => {

    return (
        <div>
            <h1>HeroesScreen</h1>
            <hr/>
            <HeroAll/>
        </div>
    );
};