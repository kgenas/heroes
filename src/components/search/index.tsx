import { useLocation, useNavigate } from "react-router-dom"
import queryString from 'query-string'; //*  npm install query-string
import { useState, useMemo } from 'react';
import { getHeroesByName } from '../../selectors/getHeroesByName';
import { HeroCard } from "../hero/HeroCard";
export const SearchScreen = () => {

    const navigate = useNavigate();
    const location = useLocation();
    
    const { q ='' } = queryString.parse(location.search);
    const [ values ,setValues] = useState({ searchText:q});
    const { searchText } = values;
    const handleInputChange =({ target }:any ) =>{

        setValues({
            ...values,
            [ target.name ]: target.value
        });
    }

    const heroesFilter = useMemo( ()=> getHeroesByName(q as string),[q]);
    const handleSearch = (e:any)=>{
        e.preventDefault();
        navigate(`?q=${ searchText }`)
    }
    return(
        <>
            <h1>Busqueda</h1>
            <hr/>
            <div className="row">
                <div className="col-5">
                    <h4>Buscar</h4>
                    <hr/>
                    <form onSubmit={ handleSearch }>
                        <input
                            type='text'
                            placeholder="Buscar un heroe"
                            className="form-control"
                            name="searchText"
                            autoComplete="off"
                            value ={ searchText as string}
                            onChange ={ handleInputChange }
                        />
                        <button
                            className="btn btn-outline-primary mt-1"
                            type='submit'
                        >
                            Buscar ....
                        </button>
                    </form>
                </div>
                <div className="col-7">
                    <h4>Resultados</h4>
                    <hr/>
                    {
                        ( q==='' ) 
                            ? <div className="alert alert-info">Buscar un heroe</div>
                            : ( heroesFilter.length === 0)
                                &&<div className="alert alert-danger">No hay resultados:{ q }</div>
                    }
                    {
                        heroesFilter.map( hero =>(
                            <HeroCard
                                key={ hero.id}
                                { ...hero }
                            />
                        ))
                    }
                </div>
            </div>
        </>
    )
}