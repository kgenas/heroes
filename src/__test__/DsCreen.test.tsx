import { renderWithProviders } from "../utils/test-utils";
import { screen } from "@testing-library/react";
import { DcScreen } from "../components/dc";

describe('DcScreen',() =>{

    test('Renders DcScreen',() =>{
 
        renderWithProviders(<DcScreen/>)
        const value = screen.getByText(/DC Screen/i);
        expect( value ).toBeInTheDocument();
        //eslint-disable-next-line          
        screen.debug();
    });
    test('DcScreen Snapshot', () =>{
       const { baseElement } = renderWithProviders(<DcScreen/>)  ;
       expect( baseElement ).toMatchSnapshot();
    })
})