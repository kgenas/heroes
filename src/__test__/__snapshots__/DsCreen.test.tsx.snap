// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DcScreen DcScreen Snapshot 1`] = `
<body>
  <div>
    <div>
      <h1>
        DC Screen
      </h1>
      <hr />
      <div
        class="row rows-cols-1 row-cols-md-3 g-3 animate__animated animate__fadeIn"
      >
        <div
          class="col animate__animated animate__fadeIn"
        >
          <div
            class="card"
          >
            <div
              class="row no-gutters"
            >
              <div
                class="col-4"
              >
                <img
                  alt="Batman"
                  class="card-img"
                  src="dc-batman.jpg"
                />
              </div>
              <div
                class="col-8"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-litle"
                  >
                    Batman
                  </h5>
                  <p
                    class="card-test"
                  >
                    Bruce Wayne
                  </p>
                  <p
                    class="card-text"
                  >
                    <small
                      class="text-muted"
                    >
                      Detective Comics #27
                    </small>
                  </p>
                  <a
                    href="/hero/dc-batman"
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="col animate__animated animate__fadeIn"
        >
          <div
            class="card"
          >
            <div
              class="row no-gutters"
            >
              <div
                class="col-4"
              >
                <img
                  alt="Superman"
                  class="card-img"
                  src="dc-superman.jpg"
                />
              </div>
              <div
                class="col-8"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-litle"
                  >
                    Superman
                  </h5>
                  <p
                    class="card-test"
                  >
                    Kal-El
                  </p>
                  <p
                    class="card-text"
                  >
                    <small
                      class="text-muted"
                    >
                      Action Comics #1
                    </small>
                  </p>
                  <a
                    href="/hero/dc-superman"
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="col animate__animated animate__fadeIn"
        >
          <div
            class="card"
          >
            <div
              class="row no-gutters"
            >
              <div
                class="col-4"
              >
                <img
                  alt="Flash"
                  class="card-img"
                  src="dc-flash.jpg"
                />
              </div>
              <div
                class="col-8"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-litle"
                  >
                    Flash
                  </h5>
                  <p
                    class="card-test"
                  >
                    Jay Garrick
                  </p>
                  <p
                    class="text-muted"
                  >
                    Jay Garrick, Barry Allen, Wally West, Bart Allen
                  </p>
                  <p
                    class="card-text"
                  >
                    <small
                      class="text-muted"
                    >
                      Flash Comics #1
                    </small>
                  </p>
                  <a
                    href="/hero/dc-flash"
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="col animate__animated animate__fadeIn"
        >
          <div
            class="card"
          >
            <div
              class="row no-gutters"
            >
              <div
                class="col-4"
              >
                <img
                  alt="Green Lantern"
                  class="card-img"
                  src="dc-green.jpg"
                />
              </div>
              <div
                class="col-8"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-litle"
                  >
                    Green Lantern
                  </h5>
                  <p
                    class="card-test"
                  >
                    Alan Scott
                  </p>
                  <p
                    class="text-muted"
                  >
                    Alan Scott, Hal Jordan, Guy Gardner, John Stewart, Kyle Raynor, Jade, Sinestro, Simon Baz
                  </p>
                  <p
                    class="card-text"
                  >
                    <small
                      class="text-muted"
                    >
                      All-American Comics #16
                    </small>
                  </p>
                  <a
                    href="/hero/dc-green"
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="col animate__animated animate__fadeIn"
        >
          <div
            class="card"
          >
            <div
              class="row no-gutters"
            >
              <div
                class="col-4"
              >
                <img
                  alt="Green Arrow"
                  class="card-img"
                  src="dc-arrow.jpg"
                />
              </div>
              <div
                class="col-8"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-litle"
                  >
                    Green Arrow
                  </h5>
                  <p
                    class="card-test"
                  >
                    Oliver Queen
                  </p>
                  <p
                    class="card-text"
                  >
                    <small
                      class="text-muted"
                    >
                      More Fun Comics #73
                    </small>
                  </p>
                  <a
                    href="/hero/dc-arrow"
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="col animate__animated animate__fadeIn"
        >
          <div
            class="card"
          >
            <div
              class="row no-gutters"
            >
              <div
                class="col-4"
              >
                <img
                  alt="Wonder Woman"
                  class="card-img"
                  src="dc-wonder.jpg"
                />
              </div>
              <div
                class="col-8"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-litle"
                  >
                    Wonder Woman
                  </h5>
                  <p
                    class="card-test"
                  >
                    Princess Diana
                  </p>
                  <p
                    class="card-text"
                  >
                    <small
                      class="text-muted"
                    >
                      All Star Comics #8
                    </small>
                  </p>
                  <a
                    href="/hero/dc-wonder"
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="col animate__animated animate__fadeIn"
        >
          <div
            class="card"
          >
            <div
              class="row no-gutters"
            >
              <div
                class="col-4"
              >
                <img
                  alt="Martian Manhunter"
                  class="card-img"
                  src="dc-martian.jpg"
                />
              </div>
              <div
                class="col-8"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-litle"
                  >
                    Martian Manhunter
                  </h5>
                  <p
                    class="card-test"
                  >
                    J'onn J'onzz
                  </p>
                  <p
                    class="text-muted"
                  >
                    Martian Manhunter
                  </p>
                  <p
                    class="card-text"
                  >
                    <small
                      class="text-muted"
                    >
                      Detective Comics #225
                    </small>
                  </p>
                  <a
                    href="/hero/dc-martian"
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="col animate__animated animate__fadeIn"
        >
          <div
            class="card"
          >
            <div
              class="row no-gutters"
            >
              <div
                class="col-4"
              >
                <img
                  alt="Robin/Nightwing"
                  class="card-img"
                  src="dc-robin.jpg"
                />
              </div>
              <div
                class="col-8"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-litle"
                  >
                    Robin/Nightwing
                  </h5>
                  <p
                    class="card-test"
                  >
                    Dick Grayson
                  </p>
                  <p
                    class="card-text"
                  >
                    <small
                      class="text-muted"
                    >
                      Detective Comics #38
                    </small>
                  </p>
                  <a
                    href="/hero/dc-robin"
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="col animate__animated animate__fadeIn"
        >
          <div
            class="card"
          >
            <div
              class="row no-gutters"
            >
              <div
                class="col-4"
              >
                <img
                  alt="Blue Beetle"
                  class="card-img"
                  src="dc-blue.jpg"
                />
              </div>
              <div
                class="col-8"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-litle"
                  >
                    Blue Beetle
                  </h5>
                  <p
                    class="card-test"
                  >
                    Dan Garret
                  </p>
                  <p
                    class="text-muted"
                  >
                    Dan Garret, Ted Kord, Jaime Reyes
                  </p>
                  <p
                    class="card-text"
                  >
                    <small
                      class="text-muted"
                    >
                      Mystery Men Comics #1
                    </small>
                  </p>
                  <a
                    href="/hero/dc-blue"
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="col animate__animated animate__fadeIn"
        >
          <div
            class="card"
          >
            <div
              class="row no-gutters"
            >
              <div
                class="col-4"
              >
                <img
                  alt="Black Canary"
                  class="card-img"
                  src="dc-black.jpg"
                />
              </div>
              <div
                class="col-8"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-litle"
                  >
                    Black Canary
                  </h5>
                  <p
                    class="card-test"
                  >
                    Dinah Drake
                  </p>
                  <p
                    class="text-muted"
                  >
                    Dinah Drake, Dinah Lance
                  </p>
                  <p
                    class="card-text"
                  >
                    <small
                      class="text-muted"
                    >
                      Flash Comics #86
                    </small>
                  </p>
                  <a
                    href="/hero/dc-black"
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`ResultModal DcScreen Snapshot 1`] = `
<body>
  <div>
    <div>
      <h1>
        DC Screen
      </h1>
      <hr />
      <div
        class="row rows-cols-1 row-cols-md-3 g-3 animate__animated animate__fadeIn"
      >
        <div
          class="col animate__animated animate__fadeIn"
        >
          <div
            class="card"
          >
            <div
              class="row no-gutters"
            >
              <div
                class="col-4"
              >
                <img
                  alt="Batman"
                  class="card-img"
                  src="dc-batman.jpg"
                />
              </div>
              <div
                class="col-8"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-litle"
                  >
                    Batman
                  </h5>
                  <p
                    class="card-test"
                  >
                    Bruce Wayne
                  </p>
                  <p
                    class="card-text"
                  >
                    <small
                      class="text-muted"
                    >
                      Detective Comics #27
                    </small>
                  </p>
                  <a
                    href="/hero/dc-batman"
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="col animate__animated animate__fadeIn"
        >
          <div
            class="card"
          >
            <div
              class="row no-gutters"
            >
              <div
                class="col-4"
              >
                <img
                  alt="Superman"
                  class="card-img"
                  src="dc-superman.jpg"
                />
              </div>
              <div
                class="col-8"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-litle"
                  >
                    Superman
                  </h5>
                  <p
                    class="card-test"
                  >
                    Kal-El
                  </p>
                  <p
                    class="card-text"
                  >
                    <small
                      class="text-muted"
                    >
                      Action Comics #1
                    </small>
                  </p>
                  <a
                    href="/hero/dc-superman"
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="col animate__animated animate__fadeIn"
        >
          <div
            class="card"
          >
            <div
              class="row no-gutters"
            >
              <div
                class="col-4"
              >
                <img
                  alt="Flash"
                  class="card-img"
                  src="dc-flash.jpg"
                />
              </div>
              <div
                class="col-8"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-litle"
                  >
                    Flash
                  </h5>
                  <p
                    class="card-test"
                  >
                    Jay Garrick
                  </p>
                  <p
                    class="text-muted"
                  >
                    Jay Garrick, Barry Allen, Wally West, Bart Allen
                  </p>
                  <p
                    class="card-text"
                  >
                    <small
                      class="text-muted"
                    >
                      Flash Comics #1
                    </small>
                  </p>
                  <a
                    href="/hero/dc-flash"
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="col animate__animated animate__fadeIn"
        >
          <div
            class="card"
          >
            <div
              class="row no-gutters"
            >
              <div
                class="col-4"
              >
                <img
                  alt="Green Lantern"
                  class="card-img"
                  src="dc-green.jpg"
                />
              </div>
              <div
                class="col-8"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-litle"
                  >
                    Green Lantern
                  </h5>
                  <p
                    class="card-test"
                  >
                    Alan Scott
                  </p>
                  <p
                    class="text-muted"
                  >
                    Alan Scott, Hal Jordan, Guy Gardner, John Stewart, Kyle Raynor, Jade, Sinestro, Simon Baz
                  </p>
                  <p
                    class="card-text"
                  >
                    <small
                      class="text-muted"
                    >
                      All-American Comics #16
                    </small>
                  </p>
                  <a
                    href="/hero/dc-green"
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="col animate__animated animate__fadeIn"
        >
          <div
            class="card"
          >
            <div
              class="row no-gutters"
            >
              <div
                class="col-4"
              >
                <img
                  alt="Green Arrow"
                  class="card-img"
                  src="dc-arrow.jpg"
                />
              </div>
              <div
                class="col-8"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-litle"
                  >
                    Green Arrow
                  </h5>
                  <p
                    class="card-test"
                  >
                    Oliver Queen
                  </p>
                  <p
                    class="card-text"
                  >
                    <small
                      class="text-muted"
                    >
                      More Fun Comics #73
                    </small>
                  </p>
                  <a
                    href="/hero/dc-arrow"
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="col animate__animated animate__fadeIn"
        >
          <div
            class="card"
          >
            <div
              class="row no-gutters"
            >
              <div
                class="col-4"
              >
                <img
                  alt="Wonder Woman"
                  class="card-img"
                  src="dc-wonder.jpg"
                />
              </div>
              <div
                class="col-8"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-litle"
                  >
                    Wonder Woman
                  </h5>
                  <p
                    class="card-test"
                  >
                    Princess Diana
                  </p>
                  <p
                    class="card-text"
                  >
                    <small
                      class="text-muted"
                    >
                      All Star Comics #8
                    </small>
                  </p>
                  <a
                    href="/hero/dc-wonder"
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="col animate__animated animate__fadeIn"
        >
          <div
            class="card"
          >
            <div
              class="row no-gutters"
            >
              <div
                class="col-4"
              >
                <img
                  alt="Martian Manhunter"
                  class="card-img"
                  src="dc-martian.jpg"
                />
              </div>
              <div
                class="col-8"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-litle"
                  >
                    Martian Manhunter
                  </h5>
                  <p
                    class="card-test"
                  >
                    J'onn J'onzz
                  </p>
                  <p
                    class="text-muted"
                  >
                    Martian Manhunter
                  </p>
                  <p
                    class="card-text"
                  >
                    <small
                      class="text-muted"
                    >
                      Detective Comics #225
                    </small>
                  </p>
                  <a
                    href="/hero/dc-martian"
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="col animate__animated animate__fadeIn"
        >
          <div
            class="card"
          >
            <div
              class="row no-gutters"
            >
              <div
                class="col-4"
              >
                <img
                  alt="Robin/Nightwing"
                  class="card-img"
                  src="dc-robin.jpg"
                />
              </div>
              <div
                class="col-8"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-litle"
                  >
                    Robin/Nightwing
                  </h5>
                  <p
                    class="card-test"
                  >
                    Dick Grayson
                  </p>
                  <p
                    class="card-text"
                  >
                    <small
                      class="text-muted"
                    >
                      Detective Comics #38
                    </small>
                  </p>
                  <a
                    href="/hero/dc-robin"
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="col animate__animated animate__fadeIn"
        >
          <div
            class="card"
          >
            <div
              class="row no-gutters"
            >
              <div
                class="col-4"
              >
                <img
                  alt="Blue Beetle"
                  class="card-img"
                  src="dc-blue.jpg"
                />
              </div>
              <div
                class="col-8"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-litle"
                  >
                    Blue Beetle
                  </h5>
                  <p
                    class="card-test"
                  >
                    Dan Garret
                  </p>
                  <p
                    class="text-muted"
                  >
                    Dan Garret, Ted Kord, Jaime Reyes
                  </p>
                  <p
                    class="card-text"
                  >
                    <small
                      class="text-muted"
                    >
                      Mystery Men Comics #1
                    </small>
                  </p>
                  <a
                    href="/hero/dc-blue"
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="col animate__animated animate__fadeIn"
        >
          <div
            class="card"
          >
            <div
              class="row no-gutters"
            >
              <div
                class="col-4"
              >
                <img
                  alt="Black Canary"
                  class="card-img"
                  src="dc-black.jpg"
                />
              </div>
              <div
                class="col-8"
              >
                <div
                  class="card-body"
                >
                  <h5
                    class="card-litle"
                  >
                    Black Canary
                  </h5>
                  <p
                    class="card-test"
                  >
                    Dinah Drake
                  </p>
                  <p
                    class="text-muted"
                  >
                    Dinah Drake, Dinah Lance
                  </p>
                  <p
                    class="card-text"
                  >
                    <small
                      class="text-muted"
                    >
                      Flash Comics #86
                    </small>
                  </p>
                  <a
                    href="/hero/dc-black"
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
